### DISCUSSION TITLE
Is disabling realtime for one table supposed to disable realtime for all other tables that have it enabled?

### DISCUSSION THREAD

**Discussion by sagitarious12** [supabase/supabase#26896] [Questions]
In my supabase instance I have realtime enabled on 4 different tables. I was working on building out a new table which needed to be realtime enabled and so I turned it on. Later, I needed to turn it off as I continued development of my feature, from the table view I clicked on the "Realtime On" button so that I could turn it off for this table. When I viewed my other tables, they still showed "Realtime On". But for each of the other tables that were still On, there were no events being triggered. Is this on purpose or should I report a bug?

**Comment by sagitarious12** [comment]
I should also mention that when I went to the table that I had previously turned off, I clicked on the "Realtime Off" button to enable realtime for this table again and it ended up Re-enabling realtime for all of the other tables that were supposed to have always been on in the first place.

**Comment by GaryAustin1** [comment]
You would probably need to show your steps in a video and file a bug.  Sounds odd as just turning them on and off on two tables (just tested) works for me.
This is the toggle I used.
![image](https://github.com/supabase/supabase/assets/54564956/2e06b109-9e79-4e0e-b050-1a335caba75d)

**Reply by sagitarious12** [reply]
would writing a step by step with images be sufficient?

**Reply by GaryAustin1** [reply]
I can't answer that.  You'll have to try it and post an issue in supabase/supabase issues.  All depends on if some time/curser thing is involved.

**Comment by sagitarious12** [comment]
step 1: have two tables or more that each have realtime enabled (in my case all of said tables are in a schema other than public)

step 2: using the js sdk subscribe to each of the tables (in my case I am listening to the INSERT events for all of them)

step 3: perform insert to verify realtime events are triggered in js code. (my code has been working for several months just fine)

step 4: go onto one table and click the following buttons: 
![image](https://github.com/supabase/supabase/assets/33358701/785d0d83-ad7f-434d-91e3-446b4cbc5fc3)
![image](https://github.com/supabase/supabase/assets/33358701/37b4d7c3-680c-4a3d-9138-bb91629425aa)

step 5: check a table that isnt the one that you disabled that is supposed to have realtime enabled and see that the UI shows that is is still enabled.

step 6: perform an insert into a table that is not the one that was just disabled

step 7: verify that there was no event triggered in the js code even though realtime is still enabled.

step 8: re enable the table that was previously disabled

step 9: perform an insert into the same table as step 5

step 10: verify that realtime is now enabled again for all of the other tables.

**Reply by GaryAustin1** [reply]
I'll just say I can't reproduce this on the public schema using js sdk and two tables on and off like you describe.

**Reply by GaryAustin1** [reply]
I also checked with one table in a different schema and no issue.
But I did notice that if you leave the js client running while this is going on you do get errors from the disabled table...
![image](https://github.com/supabase/supabase/assets/54564956/83a4226b-98a1-4dc0-bcfa-0ea192b7e868)

For me the first table still worked and then when I reenabled the 2nd table both started working again.
![image](https://github.com/supabase/supabase/assets/54564956/bc3c9565-b874-494c-8814-90ff20a0c2b8)


BUT I have two different channels for each table.  Is it possible you are getting channel errors that impact the other table because you have them on the same channel?

**Reply by sagitarious12** [reply]
It may be the channels, although I am not getting any errors at all. I have two different projects with the same setup and this issue occured for both projects. Also in my situation all of the tables are in a schema that is not the public schema. I will create a new project in my instance and just add the same 4 tables that I have realtime enabled and try to add a bit more logging around the js sdk side to see if I can get more information from my side.
